src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/ReportDiagnosticExtensions.cs

```diff
@@ -7,7 +7,7 @@
 
 namespace Microsoft.CodeAnalysis.Diagnostics
 {
-    internal static partial class Extensions
+    internal static partial class ReportDiagnosticExtensions
```

Comment:
Fix type name to match extension methods receiver type and moved existing helpers into shared layer.